LL-17:move old backend code

dampbacon opened the pull request:
Old backend code cleaned up a bit. Some minor refactoring may need to be done for consistency. The db model is still useful only really need to make additions for users and tags. I included the old sqlite db, with some hardcoded routes, so people can mess around with it before we migrate db elsewhere/make sweeping changes.

dampbacon committed:
clear-repo

dampbacon committed:
Old files moved
some minor tweaks to remove wip stuff and things that weren't relevant

Exotica0122 commented:
Why the necessity to nest the project? The root directory should be the main project not in another folder. Now you have `local-lattes-backend/local-lattes-backend`.
The whole point of separating out the repos was for this point.

Just put the files on the root folder

Exotica0122 commented on local-lattes-backend/cafeRecAPI.sln:
Should be git ignored. Don't want IDE settings on the repo. If you remove the nesting it should resolve

Exotica0122 commented on local-lattes-backend/cafeRecAPI/CafeDB/CafeDB.db:
Don't want local DB file please. Please clean up....

Exotica0122 commented on local-lattes-backend/cafeRecAPI/Controllers/CafeApiController.cs:
```diff
@@ -0,0 +1,92 @@
+        private readonly ICafeService cafeService;
+        private readonly ICafeRepo _repository;
+
+
```
Remove whitespace

```suggestion
```

Exotica0122 replied:
There's lot's of whitespace in this file. Please clean up

Exotica0122 commented on local-lattes-backend/cafeRecAPI/Controllers/CafeApiController.cs:
```diff
@@ -0,0 +1,92 @@
+        public async Task<List<Cafe>> GetCafeList(string location)
+        {
+            var cafes = await cafeService.GetCafesAsync(location);
+            return cafes;
```
Could oneline
```suggestion
            return await cafeService.GetCafesAsync(location);
```

Exotica0122 replied:
Can clean up the business logic in a separate PR. But plz clean up whitespaces

Exotica0122 commented on local-lattes-backend/cafeRecAPI/Controllers/CafeApiController.cs:
```diff
@@ -0,0 +1,92 @@
+            };
+
+            return Ok(osInfo);
+        }
```
umm why do you need this?

dampbacon replied:
was messing around with docker configuration, at some point, i will remove.

Exotica0122 replied:
Just remove it now you don't need this for docker

dampbacon commented:
it was an oversight

Exotica0122 commented on local-lattes-backend/cafeRecAPI/Data/CafeDBContext.cs:
```diff
@@ -0,0 +1,51 @@
+        public DbSet<SearchLocation> Locations { get; set; }
+        public DbSet<CafeSearchLocation> CafeSearchLocations { get; set; }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
```
Weird naming OnCafeCreated` would be better. Don't name variable/functions with "-ing" what does that mean? Does it mean it's created or it's creating? and if it's "creating" when is it finished created.

dampbacon replied:
it's the rules for any model relationship, not just cafe, unsure what else to name it

Exotica0122 commented on local-lattes-backend/cafeRecAPI/Data/CafeDBContext.cs:
```diff
@@ -0,0 +1,51 @@
+    }
+
+
+}
```
Try to add whitespace after each file or github won't like it
```suggestion
}

```

Exotica0122 commented on local-lattes-backend/cafeRecAPI/Data/CafeRepo.cs:
What's up with the indentation with this file?

Exotica0122 commented on local-lattes-backend/cafeRecAPI/Dockerfile:
Please don't add dockerfile here. This PR is scoped for cleaning up the old repo. Your increasing the scope and increasing load for code reviewers. Do it in a separate ticket and PR please

dampbacon replied:
this is from the old repo, i can remove it if you want.

Exotica0122 replied:
Yea just remove it for this PR

Exotica0122 commented on local-lattes-backend/cafeRecAPI/Utils/makeTags.cs:
What? just delete this file if it has nothing

dampbacon committed:
unnested the folder

folder structure now as created by visual studio. Double click solution file (.sln) to set up configuration correctly

Exotica0122 reviewed (commented):
Some files should be .gitignored.
Also please fix the weird indentations and whitespaces

Exotica0122 commented:
Bro you're still nesting the project. The Controllers should be at the root of the project. We want a flat folder structure
Do you understand what I mean? Right now you have a project inside a another project. Previously it was a project inside a project inside a project (3 levels deep)

dampbacon committed:
updated for comments

dampbacon commented:
forgot to remove .sln,
it's a text file that groups together multiple project files.
not really an ide configuration.
visual studio complains if i touch that or vxcproj (compilation files)

dampbacon commented:
*.csproj

Exotica0122 commented on Controllers/CafeApiController.cs:
```diff
@@ -0,0 +1,50 @@
+    public class CafeApiController : ControllerBase
+    {
+        private readonly ILogger<CafeApiController> _logger;
+        private readonly ICafeService cafeService;
```
Good standards to prefix private fields with underscore
```suggestion
        private readonly ICafeService _cafeService;
```

Exotica0122 commented on DTO/CafeDto.cs:
Is it just me or is your IDE setup a bit weird. You have 2 space indenting in some files and 4 spaces in some other

dampbacon replied:
on my ide it looks all the same, any files in particular you noticed?

Exotica0122 commented on Data/CafeRepo.cs:
```diff
@@ -0,0 +1,59 @@
+//implement
```
Do to put TODOs. Some IDEs have built in methods to parse through TODOs and find them
```suggestion
// @TODO: description...
```

dampbacon replied:
ill go throught tomorrow and edit the relevant comments

dampbacon committed:
Update Controllers/CafeApiController.cs

Co-authored-by: Peter An <75343007+Exotica0122@users.noreply.github.com>

dampbacon committed:
finished rename

dampbacon commented:
Somthing to note is that the program does not compile in it's current state due to missing db directory (not the db).
Hopefully all you need to do is modify the connection so it creates a db based on the existing models once setup with the remote thing.

Further more onto the spacing thing, not sure if you saw above but it looks the same in my IDE. Only thing i see it kinda weird is that it looks different here on github. I'll modify the files i see here.

dampbacon committed:
Update CafeDBContext.cs spacing

dampbacon committed:
plz work

dampbacon commented:
fuck changing the spacing here then commiting reverts it to 8 spaces for some reason

Exotica0122 reviewed (approved):
Gucci for now, merge it when you are ready
